web/src/providers/Test/hooks/useTestCrud.ts

```diff
@@ -28,11 +36,11 @@ const useTestCrud = () => {
   const {onOpen} = useMissingVariablesModal();
 
   const runTest = useCallback(
-    async (test: Test, runId?: string, environmentId = selectedEnvironment?.id) => {
-      const run = async (variables: TEnvironmentValue[] = []) => {
+    async ({test, environmentId = selectedEnvironment?.id, variables = [], onCancel = noop}: TTestRunRequest) => {
```

Comment:
This was getting out of hand, that's why I moved it to an object as it has better handling on optional vs required params